pkg/bgpv1/manager/reconcile.go

```diff
@@ -151,6 +161,11 @@ func (r *PreflightReconciler) Reconcile(ctx context.Context, p ReconcileParams)
 				AdvertiseInactiveRoutes: true,
 			},
 		},
+		OnFIBEvent: func() {
+			if r.signaler != nil {
+				r.signaler.Event(struct{}{})
```

Comment:
I wonder if we shouldn't have some option for this instead of signaling on each event unconditionally? Too many neighbors and/or a flapping neighbor may trigger too many unnecessary resyncs.

Alternatively (just brain-storming in my head), maybe it would make sense to pass the actual `Path` that triggered the event and define a `ConfigReconciler` API method that would be called on FIB event on all reconcilers - and they could decide whether the event is worth triggering a reconciliation?

Reply:
The latter approach sounds the same as just triggering reconciliation at the high level, and each reconciler determining that they need to do no work. 

I think we need to design all reconcilers to do minimal work to understand if they can just exit, which i think is the case today. 

The former question... its possible we could add a method to a Reconciler like:
'NeedsReconcilationOnFIBEvent bool'. 

We would then need a way to ask the BGP-CP, are we running with any ConfigReconcilers which need reconciliation on fib events. hmm. Something like this?

Reply:
> We would then need a way to ask the BGP-CP, are we running with any ConfigReconcilers which need reconciliation on fib events. hmm. Something like this?

yeah, something like that could work as well?

My original idea was something like:

```go
type ConfigReconciler interface {
	Priority() int
	Reconcile(ctx context.Context, params ReconcileParams) error
	FIBEvent(p *types.Path)
}
```

Then I guess most reconcilers would just return in `FIBEvent` and some could do something like:

```go
func (r *AwesomeReconciler) FIBEvent(p *types.Path) {
  if p.NLRI.AFI() == x &&  p.NLRI.SAAFI() == y {
     signal_to_trigger_reconciliation()
  }
}
```

That way we could trigger reconciliation of all reconcilers only upon relevant FIB events..